app/controllers/api/v1/tags_controller.rb

```diff
@@ -18,6 +18,38 @@ def show
 
         render :json => tag
       end
+
+      # Present these as tags
+      private_class_method def self.api_doc_definition
+        @api_doc_definition ||= api_doc.definitions["Tag"]
+      end
+
+      def self.presentation_name
+        "Tag".freeze
+      end
+
+      def tag
+        primary_instance = primary_collection_model.find(request_path_parts["primary_collection_id"])
```

Comment:
Wondering if the tag / untag logic should be moved to a service.  This controller is heavy with a lot of implementation logic, at least more than we normally use in a controller.

Reply:
We're probably going to keep it here for now since the end-goal is to have this controller/model in the common gem with the new "opt-in" feature that would pull it in for us. We're just mirroring topo for now (as is sources) so that when we do need to switch the implementation out nothing should drastically change for us.